sdk-http-vertx/src/main/java/dev/restate/sdk/http/vertx/RequestHttpServerHandler.java

```diff
@@ -152,22 +177,82 @@ private Executor currentContextExecutor(Context currentContext) {
   }
 
   private void handleDiscoveryRequest(HttpServerRequest request) {
-    // Compute response and write it back
-    DeploymentManifestSchema response = this.restateEndpoint.handleDiscoveryRequest();
-    Buffer responseBuffer;
-    try {
-      responseBuffer = Buffer.buffer(MANIFEST_OBJECT_MAPPER.writeValueAsBytes(response));
-    } catch (JsonProcessingException e) {
-      LOG.warn("Error when writing out the manifest POJO", e);
-      request.response().setStatusCode(INTERNAL_SERVER_ERROR.code()).end();
-      return;
+    final String accept = request.getHeader(ACCEPT);
+
+    final Discovery.ServiceDiscoveryProtocolVersion serviceDiscoveryProtocolVersion =
+        selectSupportedServiceDiscoveryProtocolVersion(accept);
+
+    if (serviceDiscoveryProtocolVersion == Discovery.ServiceDiscoveryProtocolVersion.V1) {
+      // Compute response and write it back
+      DeploymentManifestSchema response = this.restateEndpoint.handleDiscoveryRequest();
+
+      Buffer responseBuffer;
+      try {
+        responseBuffer = Buffer.buffer(MANIFEST_OBJECT_MAPPER.writeValueAsBytes(response));
+      } catch (JsonProcessingException e) {
+        LOG.warn("Error when writing out the manifest POJO", e);
+        request.response().setStatusCode(INTERNAL_SERVER_ERROR.code()).end();
+        return;
+      }
+
+      request
+          .response()
+          .setStatusCode(OK.code())
+          .putHeader(X_RESTATE_SERVER_KEY, X_RESTATE_SERVER_VALUE)
+          .putHeader(
+              CONTENT_TYPE,
+              serviceDiscoveryProtocolVersionToContentType(serviceDiscoveryProtocolVersion))
+          .end(responseBuffer);
+    } else {
+      request
+          .response()
+          .setStatusCode(UNSUPPORTED_MEDIA_TYPE.code())
+          .setStatusMessage(
+              "Service endpoint does not support the accepted service discovery protocol versions.")
```

Comment:
HTTP/2 doesn't support reason phrases, this message will be lost I think.

What I suggest is the following:

* `LOG.warn` this error, as it seems very much unexpected and it's ok to be more informative to users in those cases.
* Write only the status code in the response